src/Bicep.Core/TypeSystem/DeclaredTypeManager.cs

```diff
@@ -1408,21 +1408,34 @@ private DeclaredTypeAssignment GetModuleType(ModuleDeclarationSyntax syntax)
 
         private DeclaredTypeAssignment? GetObjectPropertyType(ObjectPropertySyntax syntax)
         {
-            var propertyName = syntax.TryGetKeyText();
+            var propertyName = syntax.TryGetKeyText() ?? (typeManager.GetTypeInfo(syntax.Key) as StringLiteralType)?.RawStringValue;
```

Comment:
>TryGetKeyText [](http://example.com/codeflow?start=38&length=13)

Can you add a comment explaining when TryGetKeyText() would give you a null and why we'd have to fall back to the type system/literal string?